### PR TITLE
feat(integration): implement data delivery module for AppsFlyer destination

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Incorporated transformer proxy 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/af/utils.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+const DESTINATION_NAME = "AppsFlyer";
+
+// Ref for error codes of Appsflyer: https://support.appsflyer.com/hc/en-us/articles/207034486#response-codes
+/**
+ * Maps the Destination Error Codes to Rudder Error Codes
+ * Have used Obect in order to incroporate any future error code handliing
+ */
+const destToRudderStatusMap = {
+  403: {
```
Comment (yashasvibajpai) [inline v0/destinations/af/utils.js:21]: Are these the only error codes in AF?

Comment (anantjain45823) [reply v0/destinations/af/utils.js:21]: Yes only these are available.

**Review by yashasvibajpai — DISMISSED** [review]

File: v0/destinations/af/utils.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      .build();
+  }
+};
+
+const responseHandler = destinationResponse => {
+  const { response, status } = destinationResponse;
+  processResponse({
+    dresponse: response,
+    status
+  });
```
Comment (ItsSudip) [inline v0/destinations/af/utils.js:74]:
```suggestion
  const { response, status } = destinationResponse;
  processResponse(destinationResponse);
```

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/af/utils.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+/**
+ * @param {*} status status code provided by the destination
+ * @returns the rudder status code for the same error
+ */
+const getStatusAndCategory = status => {
+  const rudderStatus = destToRudderStatusMap[status]
+    ? destToRudderStatusMap[status].default
+    : 500;
```
[the remark this answers is not in the document]
Comment (anantjain45823) [reply v0/destinations/af/utils.js:38]:
```
const rudderStatus = destToRudderStatusMap[status].default
```
Like this?

[the remark this answers is not in the document]
Comment (ItsSudip) [reply v0/destinations/af/utils.js:38]: If the status doesn't match the available statuses in the object we are sending 500. Can't we send whatever status we are getting from the destination?

[the remark this answers is not in the document]
Comment (anantjain45823) [reply v0/destinations/af/utils.js:38]: Yes we can . 
Updating it

File: v0/destinations/af/utils.js (hunk 2 above)
Comment (anantjain45823) [reply v0/destinations/af/utils.js:74]: object destructuring is done according to the response

**Comment by anantjain45823** [comment]
We don't need transformer proxy as error codes are general which are already handled by Rudderstack server. Also for proxyRequest and other functions we don't need any changes.
This PR was created just to make sure that we are not missing anything.

**Pull request closed without merging**